### PR TITLE
Adds support passing configuration to the dockerized app

### LINKED ISSUE #2878: Allow users to provide a custom config to the dockerized app
## Problem Definition

I just updated my rotki Docker instance to the latest, and now I'm unable to access the UI at all (`ERR_CONNECTION_TIMED_OUT`)

## Logs

I can't show debug logs as I can't get into the UI.
`$HOME/.rotki/logs/rotki.log`:
```
[08/05/2021 19:38:46 UTC] INFO rotkehlchen.api.server: Rotki API server is running at: 0.0.0.0:4242
[08/05/2021 19:39:15 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:39:15 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.002098
[08/05/2021 19:39:45 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:39:45 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001365
[08/05/2021 19:40:16 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:40:16 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001411
[08/05/2021 19:40:46 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:40:46 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001371
[08/05/2021 19:41:16 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:41:16 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001422
[08/05/2021 19:41:47 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:41:47 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001365
[08/05/2021 19:42:17 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:42:17 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001661
[08/05/2021 19:42:48 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:42:48 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001390
[08/05/2021 19:43:18 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:43:18 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001448
[08/05/2021 19:43:48 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:43:48 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001344
[08/05/2021 19:44:19 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:44:19 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001342
[08/05/2021 19:44:49 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:44:49 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001375
[08/05/2021 19:45:20 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:45:20 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001341
[08/05/2021 19:45:50 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:45:50 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001357
[08/05/2021 19:46:20 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:46:20 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001534
[08/05/2021 19:46:51 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:46:51 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.002069
[08/05/2021 19:47:21 UTC] DEBUG rotkehlchen.api.rest: Request successful response={"result": {}, "message": ""}, status_code=200
[08/05/2021 19:47:21 UTC] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "GET /api/1/users HTTP/1.0" 200 236 0.001344
```

This simply repeats over and over, and I can't get into the UI. Is it possible to handle `rotki_config.json` via the Docker image to provide more information?

Interesting to note that the interval is 30s...

### System Description
amd64 Linux host

Rotki version: 1.16.2

### PR DESCRIPTION
**Pull request by kelsos**
Closes #2878 

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by yabirgb** [comment]
The linter is not checking the new python file and it has some comments about style that we follow in the repo

```
(env) ➜  rotki git:(docker-config) ✗ flake8 packaging/docker/entrypoint.py                                   
packaging/docker/entrypoint.py:43:59: C812 missing trailing comma
packaging/docker/entrypoint.py:72:64: C812 missing trailing comma
(env) ➜  rotki git:(docker-config) ✗ mypy packaging/docker/entrypoint.py 
packaging/docker/entrypoint.py:16: error: Function is missing a return type annotation
packaging/docker/entrypoint.py:31: error: Function is missing a return type annotation
packaging/docker/entrypoint.py:47: error: Function is missing a return type annotation
packaging/docker/entrypoint.py:48: error: Call to untyped function "load_config_from_env" in typed context
packaging/docker/entrypoint.py:49: error: Call to untyped function "load_config_from_file" in typed context
packaging/docker/entrypoint.py:105: error: Call to untyped function "load_config" in typed context
Found 6 errors in 1 file (checked 1 source file)
```

File: docs/installation_guide.rst [hunk 1]
```diff
@@ -166,6 +166,49 @@ At this point the rotki docker container should be running and you
...
+
+.. code-block:: javascript
+
+    {
+       "loglevel": "info",
+       "logfromothermodules": true,
+       "sleep-secs": 22,
+       "max_size_in_mb_all_logs": 550,
+       "max_logfiles_num": 3
+    }
```
Comment (yabirgb) [inline docs/installation_guide.rst:194]: I guess this works as the list of things that can be modified. Maybe we can add a note saying that all the options are represented above?

File: packaging/docker/entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    args = [
+        '--data-dir',
+        '/data',
+        '--logfile',
+        '/logs/rotki.log',
+        '--loglevel',
+        loglevel if loglevel is not None else DEFAULT_LOG_LEVEL
+    ]
+
+    if logfromothermodules is not None and logfromothermodules is True:
```
Comment (yabirgb) [inline packaging/docker/entrypoint.py:75]:
```suggestion
    if logfromothermodules is True:
```

File: packaging/docker/entrypoint.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+logger.info('starting nginx')
+
+nginx = subprocess.Popen('nginx -g "daemon off;"', shell=True)
+
+if nginx.returncode == 1:
+    logger.error('Failed to start nginx')
+    exit(1)
+
+time.sleep(60)
```
Comment (yabirgb) [inline packaging/docker/entrypoint.py:124]: Why is this here?

File: packaging/docker/entrypoint.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    if rotki.poll() is not None:
+        logger.error('rotki has terminated exiting')
+        exit(1)
+
+    if nginx.poll() is not None:
+        logger.error('nginx was not running')
+        exit(1)
+
+    logger.info('OK: processes still running')
+    time.sleep(60)
```
Comment (yabirgb) [inline packaging/docker/entrypoint.py:136]: Also you can avoid the `time.sleep(60)` in 124 by putting this at the beginning of the loop

File: packaging/docker/entrypoint.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        'max_logfiles_num': max_logfiles_num,
+        'max_size_in_mb_all_logs': max_size_in_mb_all_logs
+    }
+
+
+def load_config():
+    env_config = load_config_from_env()
+    file_config = load_config_from_file()
+
+    if file_config is not None:
```
Comment (yabirgb) [inline packaging/docker/entrypoint.py:51]: This makes that the two methods can't be used at the same time. Is that intentional?

File: docs/installation_guide.rst [hunk 6]
```diff
@@ -166,6 +166,49 @@ At this point the rotki docker container should be running and you
...
+You need to create a ``rotki_config.json`` file and put it inside ``$HOME/.rotki/config`` directory.
+
+.. code-block:: javascript
+
+    {
+       "loglevel": "info",
+       "logfromothermodules": true,
+       "sleep-secs": 22,
+       "max_size_in_mb_all_logs": 550,
+       "max_logfiles_num": 3
```
Comment (yabirgb) [inline docs/installation_guide.rst:193]: You mix here three name formats. Is better to stick with one. For python variables also is better `log_from_other_modules` in the python code

Comment (kelsos) [reply docs/installation_guide.rst:193]: I am aware but the `rotki_config.json` has had that format for a long time now.

Comment (kelsos) [reply docs/installation_guide.rst:193]: This is not something that was introduced right now, I just reused the existing settings format.

File: packaging/docker/entrypoint.py (hunk 5 above)
Comment (kelsos) [reply packaging/docker/entrypoint.py:51]: Yes, it is intentional, do you think it would make sense to allow mix and match? with the file overwriting any other settings?

File: docs/installation_guide.rst (hunk 1 above)
Comment (kelsos) [reply docs/installation_guide.rst:194]: So the comment bellow is not sufficient, sure I will reword

**Review by yabirgb — APPROVED** [review]
Noticed that we don't do validation of the arguments but this is properly logged in the docker logs that is the maximum we will achieve doing it so LGTM
